src/terminal/terminal/terminal.h

```diff
@@ -547,6 +578,88 @@ struct guac_terminal {
 
 };
 
+/**
+ * Configuration options that may be passed when creating a new guac_terminal.
+ *
+ * Note that guac_terminal_options should not be instantiated directly -
+ * to create a new options struct, use guac_terminal_options_create.
+ */
+typedef struct guac_terminal_options {
+
+    /**
+     * The client to which the terminal will be rendered.
+     */
+    guac_client* client;
+
+    /**
+     * The guac_common_clipboard which will contain the current clipboard
+     * state. It is expected that this clipboard instance will be updated
+     * both internally by the terminal and externally through received
+     * clipboard instructions. This clipboard will not be automatically
+     * freed when this terminal is freed.
+     */
+    guac_common_clipboard* clipboard;
```

Comment:
My late feedback:

The actual storage and implementation of the clipboard doesn't feel like an option. `disable_copy`, etc. definitely does, but then the allocation/handling of the clipboard implementation with respect to that option should be internal to the main `guac_terminal` struct, I'd think.